crates/jsonrpc-types/src/godwoken.rs

```diff
@@ -1187,8 +1205,91 @@ pub struct BackendInfo {
     pub validator_code_hash: H256,
     pub generator_code_hash: H256,
     pub validator_script_type_hash: H256,
+    pub backend_type: BackendType,
+}
+
+#[derive(Clone, Serialize, Deserialize, PartialEq, Eq, Hash, Debug)]
+#[serde(rename_all = "snake_case")]
+pub enum BackendType {
+    Unknown,
+    Meta,
+    Sudt,
+    Polyjuice,
+    EthAddrReg,
+}
+
+impl Default for BackendType {
+    fn default() -> Self {
+        BackendType::Unknown
+    }
 }
 
+#[derive(Clone, Serialize, Deserialize, PartialEq, Eq, Hash, Debug, Default)]
+#[serde(rename_all = "snake_case")]
+pub struct GwScript {
+    pub type_hash: H256,
+    pub script: Script,
+    pub script_type: GwScriptType,
+}
+
+#[derive(Clone, Serialize, Deserialize, PartialEq, Eq, Hash, Debug)]
+#[serde(rename_all = "snake_case")]
+pub enum GwScriptType {
+    Unknown,
+    Deposit,
+    Withdraw,
+    StateValidator,
+    StakeLock,
+    CustodianLock,
+    ChallengeLock,
+    L1Sudt,
+    L2Sudt,
+    OmniLock,
+}
+
+impl Default for GwScriptType {
+    fn default() -> Self {
+        GwScriptType::Unknown
+    }
+}
+
+#[derive(Clone, Serialize, Deserialize, PartialEq, Eq, Hash, Debug, Default)]
+#[serde(rename_all = "snake_case")]
+pub struct RollupCell {
```

Comment:
```suggestion
pub struct RollupTypeScript {
```

Reply:
I think it can be included in `NodeInfo::gw_scripts`.

Reply:
I tend to keep the original name `RollupCell`: (1. rollup type script hash is the most frequently used config for developers, so I think it is easier to find. (2. it is used to locate the rollup cell, so `RollupCell` looks intuitively to me

Reply:
yes, I agree that `RollupCell` is more intuitive. If we keep this name, then I think the `RollupCell.script` should be renamed to `RollupCell.type_script` to distinguish between lock script and type script.

Reply:
agree. `RollupCell.type_script` updated.